codelab/JinghangCodeLab/MagicNumberConverter.java

```diff
@@ -0,0 +1,41 @@
+package JinghangCodeLab;
+
+public class MagicNumberConverter {
+
+    public static int binaryToDecimal(String binary)
+    {  
+        //edge cases
+        if (binary == "0"){
+            return 0;
+        }
+
+        //negative
+        int isNegative = -1;
+        if(Integer.parseInt(binary) < 0){
```

Comment:
It's great that you thought of handling negative numbers, however, using "Integer.parseInt" may cause overflow issue, for example if the binary has 20 characters like this:  "11111111111111111111". You can try to check if the first character of the input binary is "-".